### PR TITLE
Add support multi-level inheritance of sealed

### PR DESCRIPTION
**Pull request by ctmanjak**
## Summary
Add support for multi-level inheritance of sealed class and sealed interface.

## How Has This Been Tested?
Add a new test that adds an intermediate inherited class and interface to an existing test.

### CODE REVIEW

File: fixture-monkey-api/src/main/java17/com/navercorp/fixturemonkey/api/generator/SealedTypeObjectPropertyGenerator.java [hunk 1]
```diff
@@ -55,4 +57,13 @@ public ObjectProperty generate(ObjectPropertyGeneratorContext context) {
...
 	}
+
+	private static Stream<Class<?>> collectPermittedSubclasses(Class<?> type) {
+		if (type.isSealed()) {
+			return Stream.of(type.getPermittedSubclasses())
+				.flatMap(SealedTypeObjectPropertyGenerator::collectPermittedSubclasses);
+		} else {
+			return Stream.of(type);
+		}
+	}
```
Comment (seongahjo) [inline fixture-monkey-api/src/main/java17/com/navercorp/fixturemonkey/api/generator/SealedTypeObjectPropertyGenerator.java:68]: I think it is better to return `Set` in this case.

File: fixture-monkey-tests/java-17-tests/src/test/java/com/navercorp/fixturemonkey/tests/java17/MultiLevelSealedClassTest.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+		@ConstructorProperties("value")
+		public SealedClassImpl(String value) {
+			this.value = value;
+		}
+
+		public String getValue() {
+			return value;
+		}
+	}
```
Comment (seongahjo) [inline fixture-monkey-tests/java-17-tests/src/test/java/com/navercorp/fixturemonkey/tests/java17/MultiLevelSealedClassTest.java:54]: I think it is time to add a `SealedClassTestSpecs` for test classes.

For readability, it is better to put all classes in `SealedClassTest', `MultiLevelSealedClassTest' in `SealedClassTestSpecs'.

Comment (ctmanjak) [reply fixture-monkey-tests/java-17-tests/src/test/java/com/navercorp/fixturemonkey/tests/java17/MultiLevelSealedClassTest.java:54]: Or could we just have MultiLevelSealedClassTest without SealedClassTest?

File: fixture-monkey-api/src/main/java17/com/navercorp/fixturemonkey/api/generator/SealedTypeObjectPropertyGenerator.java [hunk 3]
```diff
@@ -38,7 +40,7 @@ public ObjectProperty generate(ObjectPropertyGeneratorContext context) {
 		double nullInject = context.getNullInjectGenerator().generate(context);
 
 		Map<Property, List<Property>> childPropertiesByProperty = new HashMap<>();
-		for (Class<?> subClass : actualType.getPermittedSubclasses()) {
+		for (Class<?> subClass : collectPermittedSubclasses(actualType)) {
```
Comment (seongahjo) [inline fixture-monkey-api/src/main/java17/com/navercorp/fixturemonkey/api/generator/SealedTypeObjectPropertyGenerator.java:43]: it is better to assign `collectPermittedSubclasses(actualType)` as a variable

File: fixture-monkey-api/src/main/java17/com/navercorp/fixturemonkey/api/generator/SealedTypeObjectPropertyGenerator.java [hunk 4]
```diff
@@ -55,4 +57,20 @@ public ObjectProperty generate(ObjectPropertyGeneratorContext context) {
...
+
+	private static void collectPermittedSubclasses(Class<?> type, Set<Class<?>> subclasses) {
+		if (type.isSealed()) {
+			for (Class<?> subclass : type.getPermittedSubclasses()) {
+				collectPermittedSubclasses(subclass, subclasses);
+			}
+		} else {
+			subclasses.add(type);
+		}
+	}
```
Comment (seongahjo) [inline fixture-monkey-api/src/main/java17/com/navercorp/fixturemonkey/api/generator/SealedTypeObjectPropertyGenerator.java:75]: I think there is no need to divide it into two methods.

**Comment by seongahjo** [comment]
Please update a release note in `docs/content/v1.0.x/release-notes`, `docs/content/v1.0.x-kor/release-notes`

File: fixture-monkey-api/src/main/java17/com/navercorp/fixturemonkey/api/generator/SealedTypeObjectPropertyGenerator.java (hunk 4 above)
Comment (ctmanjak) [reply fixture-monkey-api/src/main/java17/com/navercorp/fixturemonkey/api/generator/SealedTypeObjectPropertyGenerator.java:75]: If we combine the methods into one, won't that create unnecessary HashSet instance?

Comment (seongahjo) [reply fixture-monkey-api/src/main/java17/com/navercorp/fixturemonkey/api/generator/SealedTypeObjectPropertyGenerator.java:75]: Ah, I see. I did not see the line 70.

It is better to change the second `collectPermittedSubclasses` to `doCollectPermittedSubclasses` for readability.

**Review by seongahjo — APPROVED** [review]
Great 👍  Thank you for contributing!

**Pull request merged**